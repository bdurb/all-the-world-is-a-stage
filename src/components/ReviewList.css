
.reviews {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  margin-bottom: 20px;
}

h1 {
  font-family: cursive;
  font-size: 2.8rem;
  color: whitesmoke;
  text-shadow: black 0.12em 0.1em 0.1em;
  font-weight: bold;
}