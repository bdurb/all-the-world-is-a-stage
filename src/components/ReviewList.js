import React, { Component } from 'react';
import Review from './Review';
import axios from 'axios';

import './ReviewList.css'

class ReviewList extends Component {
  state = {
    results: []
  }

  componentDidMount() {
    axios.get('http://shakespeare.podium.co/api/reviews', {
      headers: { authorization: 'koOheljmQX' }
    })
    .then(res => {
      this.setState({results: res.data.data})
    })
    .catch(err => {
      console.log(err)
    })
  }

  handleLowestSort = () => {
    let resultsCopy = [...this.state.results]
    resultsCopy.sort((a, b) => {
      return a.rating - b.rating
    })
    this.setState({results: resultsCopy})
  }

  handleHighestSort = () => {
    let resultsCopy = [...this.state.results]
    resultsCopy.sort((a, b) => {
      return b.rating - a.rating
    })
    this.setState({results: resultsCopy})
  }

  render() { 
    return (
      <div className='reviews'>
        <h1>No Legacy is so rich as honesty</h1>
        <h2>A review site dedicated to the works of Shakespeare</h2>
        <div className='buttons'>
          <button onClick={this.handleLowestSort}>Lowest</button>
          <button onClick={this.handleHighestSort}>Highest</button>
        </div>
        {this.state.results.map(review => (
          <Review review={review} key={review.id} />
        ))}
      </div>
    );
  }
}
 
export default ReviewList;