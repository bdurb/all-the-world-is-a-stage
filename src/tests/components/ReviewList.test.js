import React from 'react';
import { shallow } from 'enzyme';
import ReviewList from '../../components/ReviewList';

test('should render ReviewList correctly', () => {
  const wrapper = shallow(<ReviewList />)
  expect(wrapper).toMatchSnapshot()
})
