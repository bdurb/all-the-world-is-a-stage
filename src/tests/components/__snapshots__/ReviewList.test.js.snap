// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ReviewList correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReviewList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          No Legacy is so rich as honesty
</h1>,
        <h2>
          A review site dedicated to the works of Shakespeare
</h2>,
        <div
          className="buttons"
>
          <button
                    className="lowest-button"
                    onClick={[Function]}
          >
                    Lowest
          </button>
          <button
                    onClick={[Function]}
          >
                    Highest
          </button>
</div>,
        Array [],
      ],
      "className": "reviews",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No Legacy is so rich as honesty",
        },
        "ref": null,
        "rendered": "No Legacy is so rich as honesty",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "A review site dedicated to the works of Shakespeare",
        },
        "ref": null,
        "rendered": "A review site dedicated to the works of Shakespeare",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="lowest-button"
              onClick={[Function]}
>
              Lowest
</button>,
            <button
              onClick={[Function]}
>
              Highest
</button>,
          ],
          "className": "buttons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lowest",
              "className": "lowest-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Lowest",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Highest",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Highest",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            No Legacy is so rich as honesty
</h1>,
          <h2>
            A review site dedicated to the works of Shakespeare
</h2>,
          <div
            className="buttons"
>
            <button
                        className="lowest-button"
                        onClick={[Function]}
            >
                        Lowest
            </button>
            <button
                        onClick={[Function]}
            >
                        Highest
            </button>
</div>,
          Array [],
        ],
        "className": "reviews",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No Legacy is so rich as honesty",
          },
          "ref": null,
          "rendered": "No Legacy is so rich as honesty",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "A review site dedicated to the works of Shakespeare",
          },
          "ref": null,
          "rendered": "A review site dedicated to the works of Shakespeare",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="lowest-button"
                onClick={[Function]}
>
                Lowest
</button>,
              <button
                onClick={[Function]}
>
                Highest
</button>,
            ],
            "className": "buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lowest",
                "className": "lowest-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Lowest",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Highest",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Highest",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
